mock-certify-plugin/src/main/java/io/mosip/certify/util/CSVReader.java

```diff
@@ -19,75 +19,45 @@
+            }
 
-                // Process each record
-                for (CSVRecord record : csvParser) {
-                    String identifier = record.get(identifierColumn);
-                    Map<String, String> rowData = new HashMap<>();
-                    // Store only the configured fields
-                    for (String header : headers) {
-                        if (fieldsToInclude.contains(header) || header.equals(identifierColumn)) {
-                            rowData.put(header, record.get(header));
-                        }
+            // Process each record
+            for (CSVRecord record : csvParser) {
+                String identifier = record.get(identifierColumn);
+                JSONObject jsonObject = new JSONObject();
+                // Store only the configured fields
+                for (String header : headers) {
+                    if (dataColumns.contains(header) || header.equals(identifierColumn)) {
+                        jsonObject.put(header, record.get(header));
                     }
-
-                    // Add to dataMap
-                    dataMap.computeIfAbsent(identifier, k -> new ArrayList<>()).add(rowData);
                 }
-            } catch (IOException e) {
-                log.error("Error finding csv file path", e);
-                throw new IOException("Unable to find the CSV file.");
+
+                // Add to dataMap
+                dataMap.computeIfAbsent(identifier, k -> new ArrayList<>()).add(jsonObject);
             }
         } catch (IOException e) {
-            log.error("Error fetching csv file from classpath resource", e);
-            throw new IOException("Unable to find the classpath resource for csv file.");
+            log.error("Error finding csv file path", e);
+            throw new IOException("Unable to find the CSV file.");
         }
     }
 
     public JSONObject getJsonObjectByIdentifier(String identifier) throws DataProviderExchangeException, JSONException {
-        JSONObject jsonObject = new JSONObject();
-        List<Map<String, String>> records = dataMap.get(identifier);
+        List<JSONObject> records = dataMap.get(identifier);
         if(records == null || records.isEmpty()) {
             log.error("No identifier found.");
             throw new DataProviderExchangeException("No record found in csv with the provided identifier");
         }
-        if (records != null && !records.isEmpty()) {
-            Map<String, String> record = records.get(0);
-            // Add only configured fields to JsonObject
-            for (Map.Entry<String, String> entry : record.entrySet()) {
-                if (fieldsToInclude.contains(entry.getKey()) || entry.getKey().equals(identifierColumn)) {
-                    jsonObject.put(entry.getKey(), entry.getValue());
-                }
-            }
-        }
-        return jsonObject;
+
+        return records.get(0);
```

Comment:
when you need just the item zero always, why not use the datatype as below ?

private Map<String, JSONObject> dataMap = new HashMap<>();

Reply:
Changed the datamap to above map type